src/python/gem5/utils/looppoint.py

```diff
@@ -0,0 +1,219 @@
+# Copyright (c) 2022 The Regents of the University of California
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met: redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer;
+# redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution;
+# neither the name of the copyright holders nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from m5.util import fatal
+from pathlib import Path
+from typing import List
+from gem5.components.processors.abstract_core import AbstractCore
+from m5.objects.LoopPointManager import LoopPointManager
+import csv
+import re
+from abc import ABCMeta
+        
+        
+class BaseLoopPoints:
+    
+    __metaclass__ = ABCMeta
+    
+    def __init__(
+        self,
+        checkpointPC: List[int],
+        checkpointCount: List[int],
+        relativePC: List[int],
+        relativeCount: List[int],
+        regionID: List[int]
+    ) -> None:
+        
+        self._checkpointPC = checkpointPC
+        self._checkpointCount = checkpointCount
+        self._relativePC = relativePC
+        self._relativeCount = relativeCount
+        self._regionID = regionID
+        
+        self._looppointManager = LoopPointManager()
+    
+    def setup_cpu(
+        self, 
+        cpuList: List[AbstractCore], 
+        ) -> None:
+        self._looppointManager.target_count = self._checkpointCount
+        self._looppointManager.target_pc = self._checkpointPC
+        self._looppointManager.region_id = self._regionID
+        self._looppointManager.relative_pc = self._relativePC
+        self._looppointManager.relative_count = self._relativeCount
+        self._looppointManager.setup(cpuList)
```

Comment:
```suggestion
    def setup_processor(
        self, 
        processor: List[AbstractProcessor], 
        ) -> None:
        self._looppointManager.target_count = self._checkpointCount
        self._looppointManager.target_pc = self._checkpointPC
        self._looppointManager.region_id = self._regionID
        self._looppointManager.relative_pc = self._relativePC
        self._looppointManager.relative_count = self._relativeCount
        for core in processor.get_cores():
            core.addLoopPointProbe(self._checkpointPC + self._relativePC, self._looppointManager)
```

Note, I'm not very confident on the last line.

Whether you go with this suggestion or something closer to the original... I suggest using the name "core" or "processor" and avoiding the name "cpu" for variables.